MergeSharpWebAPI/Program.cs

```diff
@@ -93,26 +97,46 @@ private static async Task Main(string[] args)
             var serializedLwwSet = JsonConvert.SerializeObject(myLWWSetService.Get(1));
 
             var requestData = new StringContent(serializedLwwSet, Encoding.UTF8, "application/json");
-            //string myContent = await requestData.ReadAsStringAsync();
-            //Console.WriteLine("im porinting data");
-            //Console.WriteLine(myContent);
 
             Console.WriteLine("Sending Put Request for front-end");
             var result = await client.PutAsync(
                 "https://localhost:7009/LWWSet/SendLWWSetToFrontEnd", requestData);
             Console.WriteLine(result);
         });
+    }
 
+    private static async void connectionStart()
+    {
         // Start the connection
-        try
+        while (connection.State == HubConnectionState.Disconnected)
         {
-            await connection.StartAsync();
-            Console.WriteLine("Connection started");
-        }
-        catch (Exception ex)
-        {
-            Console.WriteLine("Error occured when connecting to server:");
-            Console.WriteLine(ex.Message);
+            try
+            {
+                await connection.StartAsync();
+                Console.WriteLine("Connection started");
+                await MergeSharpWebAPI.Globals.connection.InvokeAsync("SendEncodedMessage", myLWWSetService.Get(1).LwwSet.GetLastSynchronizedUpdate().Encode());
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("Error occured when connecting to server:");
+                Console.WriteLine(ex.Message);
+                Console.WriteLine(connection.State);
+                await Task.Delay(5000);
+            }
         }
     }
+    private static void Main(string[] args)
+    {
+        Thread serverThread = new Thread(StartServer);
+        serverThread.Start();
+
+        connectionReconnected();
```

Comment:
I think `ConfigureConnectionReconnected` would be a better name. Similar for the next two functions